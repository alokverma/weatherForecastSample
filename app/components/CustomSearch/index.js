import PropTypes from 'prop-types'
import React from 'react'
import {TextInput, View, Button} from 'react-native';
import s from './styles'

const CustomSearch = (props) => {
  const {value, onChange, onButtonPress} = props

  return (
    <View style={s.container}>
     
      <TextInput
        style={s.textInput}
        value={value}
        keyboardShouldPersistTaps={false}
        underlineColorAndroid="white"
        placeholderTextColor="gray"
        onChangeText={(value)=>onChange(value)}
        placeholder="Search" />
      {!!value && value.length !== 0 ? (
        <Button
          style={s.button}
          title = 'search'
          onPress={() => onButtonPress()}>
        </Button>
      ) : (<View style={s.button} />)}
    </View>
  )
}

CustomSearch.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
  onButtonPress: PropTypes.func
}

export default CustomSearch
