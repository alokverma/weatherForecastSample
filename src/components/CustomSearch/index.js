import PropTypes from 'prop-types'
import React from 'react'
import {TextInput, View, Button, TouchableOpacity} from 'react-native';
import s from './styles'
import Icon from 'react-native-vector-icons/FontAwesome';
import {THEMES} from '../../utils/constants'
const {colors} = THEMES.weatherDefaultTheme;

//Icon.loadFont();

const CustomSearch = (props) => {
  const {value, onChange, onButtonPress} = props

  return (
    <View style={s.container}>
     
      <TextInput
        style={s.textInput}
        value={value}
        keyboardShouldPersistTaps={false}
        underlineColorAndroid="white"
        keyboardType='numeric'
        placeholderTextColor="gray"
        maxLength={5}
        onChangeText={(value)=>onChange(value)}
        placeholder="Enter zip code" />
          
      {!!value && value.length !== 0 ? (
        <TouchableOpacity
            style={s.button}>
             <Icon
              name="search" 
             size={30}
              color={colors.dark} 
              onPress={() => onButtonPress()}/>
        </TouchableOpacity>
        // <Icon.Button
        //   style={s.button}
        //   icon = "search"
        //   title = 'search'
        //   onPress={() => onButtonPress()}>
        // </Icon.Button>
       
      ) : (<View style={s.button} />)}
     
    </View>
  )
}

CustomSearch.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
  onButtonPress: PropTypes.func
}

export default CustomSearch
