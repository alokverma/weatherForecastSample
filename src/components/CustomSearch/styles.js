import {StyleSheet} from 'react-native'
const height = 48

const styles = StyleSheet.create({
  container: {
    borderRadius: 5,
    height,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 4,
    backgroundColor: 'white',
    elevation: 5
  },
  button: {
    elevation: 20,
    width: 56,
    height,
    justifyContent: 'center',
    alignItems: 'center'
  },

  textInput: {
    flex: 1,
    height,
    paddingLeft:10,
    backgroundColor: 'white',
    fontSize: 18
  }
})

export default styles
