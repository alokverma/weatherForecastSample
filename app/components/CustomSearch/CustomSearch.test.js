import React from 'react';
import {shallow} from 'enzyme';
import CustomSearch from './index'

describe('CustomSearch',()=>{
    it('should render custom search component',()=>{
        const component = shallow(<CustomSearch />)
        expect(component).toMatchSnapshot()

    });
});