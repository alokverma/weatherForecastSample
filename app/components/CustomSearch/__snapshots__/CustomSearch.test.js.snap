// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSearch should render custom search component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 5,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "margin": 4,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardShouldPersistTaps={false}
    onChangeText={[Function]}
    placeholder="Search"
    placeholderTextColor="gray"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "fontSize": 18,
        "height": 48,
        "paddingLeft": 10,
      }
    }
    underlineColorAndroid="white"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "elevation": 20,
        "height": 48,
        "justifyContent": "center",
        "width": 56,
      }
    }
  />
</View>
`;
